packages/core-data/src/private-selectors.ts

```diff
@@ -21,7 +22,7 @@ type EntityRecordKey = string | number;
  * @return The undo manager.
  */
 export function getUndoManager( state: State ) {
-	return state.undoManager;
+	return getSyncProvider().getUndoManager() ?? state.undoManager;
```

Comment:
Could we run into a case where the UndoManager used get switched between refreshes? I'd say the answer is no, and during my testing I haven't run into that but figured I'd note this.

Reply:
I do like this a lot as it means that if we don't instantiate the yjs undo manager (for non postType entities), then the default GB undo manager would be given back.